From 44383e3df4896f2d04d9141f640934d3e74e04d7 Mon Sep 17 00:00:00 2001
From: Georg Richter <georg@mariadb.com>
Date: Sun, 6 Nov 2022 13:25:10 +0100
Subject: [PATCH] Fix for MDEV-29925

Since CHECK_COMPILER_FLAG doesn't work for linker flags,
we need to check the options with CHECK_LINKER_FLAG
(which is available since CMake 3.18)
diff -wpruN '--exclude=*.orig' a~/libmariadb/plugins/auth/CMakeLists.txt a/libmariadb/plugins/auth/CMakeLists.txt
--- a~/libmariadb/plugins/auth/CMakeLists.txt	1970-01-01 00:00:00
+++ a/libmariadb/plugins/auth/CMakeLists.txt	1970-01-01 00:00:00
@@ -99,7 +99,10 @@ IF(NOT WIN32)
   INCLUDE(${CC_SOURCE_DIR}/cmake/FindGSSAPI.cmake)
   IF(GSSAPI_FOUND)
     SET(GSSAPI_SOURCES ${AUTH_DIR}/auth_gssapi_client.c ${AUTH_DIR}/gssapi_client.c ${AUTH_DIR}/gssapi_errmsg.c)
-    CHECK_C_COMPILER_FLAG(-Wl,--as-needed have_C__Wl___as_needed)
+    IF (CMAKE_VERSION VERSION_GREATER 3.18)
+      INCLUDE(CheckLinkerFlag)
+      CHECK_LINKER_FLAG(C -Wl,--as-needed have__Wl___as_needed)
+    ENDIF()
   ENDIF()
 ELSE()
   SET(GSSAPI_LIBS secur32)
@@ -116,7 +119,7 @@ IF(GSSAPI_SOURCES)
   IF(CMAKE_C_COMPILER_ID MATCHES "Clang")
     SET_SOURCE_FILES_PROPERTIES(${GSSAPI_SOURCES} PROPERTY COMPILE_FLAGS "-Wno-deprecated-declarations")
   ENDIF()
-  IF(have_C__Wl___as_needed)
+  IF(have__Wl___as_needed)
     SET_TARGET_PROPERTIES(auth_gssapi_client PROPERTIES LINK_FLAGS "-Wl,--as-needed")
   ENDIF()
 ENDIF()
